virttest/utils_test/__init__.py

```diff
@@ -2047,13 +2049,24 @@ def install(self):
         """
         To download, abstract, build and install the stress tool
         """
-        self.download_stress()
         # Install the dependencies before the tool gets installed
         if self.dependency_packages:
             if not utils_package.package_install(self.dependency_packages,
                                                  session=self.session):
                 raise exceptions.TestError("Installing dependency packages for"
                                            " %s failed" % self.stress_type)
+        if self.stress_install_from_repo and self.stress_package:
+            # Install the stress package from existing repos
+            # If succeed, no need to download stress src any more
+            if not utils_package.package_install(self.stress_package,
+                                                 session=self.session):
+                logging.debug("Fail to install stress tool via repo and "
+                              "will download source to make and install it")
```

Comment:
let us uninstall in L2086 only if the installation from repo succeeded here, so reset the `self.stress_install_from_repo` if installation fails.
```python
if not utils_package.package_install(self.stress_package,
                                     session=self.session):
    self.stress_install_from_repo = False
    logging.debug("Fail to install stress tool via repo and "
                  "will download source to make and install it")
```

Reply:
@balamuruhans well, good catch. Updated.